src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/InTableFilter.java

```diff
@@ -0,0 +1,106 @@
+package io.stargate.sgv2.jsonapi.service.operation.filters.table;
+
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.bindMarker;
+
+import com.datastax.oss.driver.api.core.metadata.schema.ColumnMetadata;
+import com.datastax.oss.driver.api.querybuilder.relation.OngoingWhereClause;
+import com.datastax.oss.driver.api.querybuilder.relation.Relation;
+import com.datastax.oss.driver.api.querybuilder.term.Term;
+import io.stargate.sgv2.jsonapi.exception.ErrorCodeV1;
+import io.stargate.sgv2.jsonapi.exception.catchable.MissingJSONCodecException;
+import io.stargate.sgv2.jsonapi.exception.catchable.ToCQLCodecException;
+import io.stargate.sgv2.jsonapi.exception.catchable.UnknownColumnException;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.TableSchemaObject;
+import io.stargate.sgv2.jsonapi.service.operation.builder.BuiltCondition;
+import io.stargate.sgv2.jsonapi.service.operation.filters.table.codecs.JSONCodecRegistries;
+import io.stargate.sgv2.jsonapi.service.operation.query.TableFilter;
+import io.stargate.sgv2.jsonapi.service.operation.query.TableFilterAnalyzedUsage;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/** API filter $in and $nin against table column. */
+public class InTableFilter extends TableFilter {
+
+  private final List<Object> arrayValue;
+
+  private final Operator operator;
+
+  public enum Operator {
+    IN,
+    // TODO NIN
+    NIN
+  }
+
+  public InTableFilter(Operator operator, String path, List<Object> arrayValue) {
+    super(path);
+    this.arrayValue = arrayValue;
+    this.operator = operator;
+  }
+
+  @Override
+  public <StmtT extends OngoingWhereClause<StmtT>> StmtT apply(
+      TableSchemaObject tableSchemaObject,
+      StmtT ongoingWhereClause,
+      List<Object> positionalValues) {
+    List<Term> bindMarkers = new ArrayList<>();
+
+    // TODO this codec part we won't do in the operation level? refer to insertOne
+    //    TableInsertValuesContainer
+    for (Object inValue : arrayValue) {
+      try {
+        var codec =
+            JSONCodecRegistries.DEFAULT_REGISTRY.codecToCQL(
+                tableSchemaObject.tableMetadata(), getPathAsCqlIdentifier(), inValue);
+        positionalValues.add(codec.toCQL(inValue));
+        bindMarkers.add(bindMarker());
+      } catch (UnknownColumnException e) {
+        throw ErrorCodeV1.TABLE_COLUMN_UNKNOWN.toApiException(e.getMessage());
+      } catch (MissingJSONCodecException e) {
+        throw ErrorCodeV1.TABLE_COLUMN_TYPE_UNSUPPORTED.toApiException(e.getMessage());
+      } catch (ToCQLCodecException e) {
+        // TODO AARON - Handle error
+        throw new RuntimeException(e);
+      }
+    }
+
+    return ongoingWhereClause.where(Relation.column(getPathAsCqlIdentifier()).in(bindMarkers));
+  }
+
+  @Override
+  public BuiltCondition get() {
+    throw new UnsupportedOperationException(
```

Comment:
can you clarify what modified means, do we mean the interface will be changed ?

Reply:
added java docs